starcluster/plugins/efs.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2015 Michael Cariaso
+#
+# This file is a plugin for StarCluster.
+#
+# This plugin is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# StarCluster is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with StarCluster. If not, see <http://www.gnu.org/licenses/>.
+
+from starcluster import clustersetup
+from starcluster.logger import log
+import boto3
+
+
+class EFSPlugin(clustersetup.DefaultClusterSetup):
+    """Add EFS support
+
+    Example config:
+
+    [plugin efs]
+    SETUP_CLASS = efs.EFSPlugin
```

Comment:
perhaps. your suggested change seems plausible, but I confess I use starcluster in a somewhat weird way, and that may contribute to the difference, but the above did seem to work fine for me. This is now part of datacratic and I think the extra eyeballs and test environments is the best way to answer the question and ensure it works correctly for a wide audience.